vortex-array/src/arrays/constant/compute/mod.rs

```diff
@@ -100,6 +106,33 @@ impl UncompressedSizeFn<&ConstantArray> for ConstantEncoding {
     }
 }
 
+impl SumFn<&ConstantArray> for ConstantEncoding {
+    fn sum(&self, array: &ConstantArray) -> VortexResult<Scalar> {
+        let sum_dtype = Stat::Sum
+            .dtype(array.dtype())
+            .ok_or_else(|| vortex_err!("Sum not supported for dtype {}", array.dtype()))?;
+        let sum_ptype = PType::try_from(&sum_dtype).vortex_expect("sum dtype must be primitive");
+
+        let scalar = array.scalar();
+
+        match_each_native_ptype!(sum_ptype, |$P| {
+            sum_constant::<$P>(scalar.as_primitive(), array.len())
+        })
+    }
+}
+
+fn sum_constant<T>(primitive_scalar: PrimitiveScalar<'_>, array_len: usize) -> VortexResult<Scalar>
+where
+    T: FromPrimitiveOrF16 + NativePType,
+    Scalar: From<T>,
+{
+    let v = primitive_scalar.as_::<T>()?;
+    let valid_count = T::from(array_len).vortex_expect("valid_count must fit the sum type");
+    let sum = v.map(|v| v * valid_count).unwrap_or_default();
```

Comment:
how do we want to handle overflow on sums?

Reply:
The semantics are that Sum should return null scalar for overflow

Reply:
fixed.